### PR TITLE
Add a flowchart environnement and its listof

### PR DESCRIPTION
**Pull request by cedounet**
So we can more easily access those flowcharts from pdf and website, rather than looking in the crowded list of figures.

### CODE REVIEW

File: book/troubleshooting/misc.tex [hunk 1]
```diff
@@ -320,14 +320,14 @@ \subsection{Why does my starter smell like vinegar or acetone?}
 When tasting acetic acid, the flavor of your bread is often perceived
 as quite strong.
 
-\begin{figure}[!htb]
+\begin{flowchart}[!htb]
 \begin{center}
   \input{figures/fig-ethanol-oxidation.tex}
   \caption[acetic acid creation]{Oxygen is required to create acetic
```
Comment (hendricius) [inline book/troubleshooting/misc.tex:326]:
```suggestion
  \caption[Acetic acid creation]{Oxygen is required to create acetic
```

**Comment by hendricius** [comment]
For some reason they are not shown in the website editioimage

I could remove the list for now on the website?

**Comment by cedounet** [comment]
That is the easy "fix"... 
- Do they show up in ebook ?
- Do they show up in html ?

**Comment by hendricius** [comment]
Good point. The CI doesn't run trying to make the ebook version. This seems to be broken too. Got this error when running `make bake`

```
[WARNING] domfilter: ...ive/2023/texmf-dist/tex/luatex/luaxml/luaxml-mod-xml.lua:175: Unbalanced Tag (/body) [char=15766]

[WARNING] domfilter: DOM parsing of listtablename.html failed:
[WARNING] domfilter: ...ive/2023/texmf-dist/tex/luatex/luaxml/luaxml-mod-xml.lua:175: Unbalanced Tag (/body) [char=15766]

[WARNING] domfilter: DOM parsing of listfigurename.html failed:
[WARNING] domfilter: ...ive/2023/texmf-dist/tex/luatex/luaxml/luaxml-mod-xml.lua:175: Unbalanced Tag (/body) [char=22667]

[WARNING] domfilter: DOM parsing of listfigurename.html failed:
[WARNING] domfilter: ...ive/2023/texmf-dist/tex/luatex/luaxml/luaxml-mod-xml.lua:175: Unbalanced Tag (/body) [char=22667]

[WARNING] exec_epub: Missing opf file
...023/texmf-dist/scripts/tex4ebook/tex4ebook-exec_epub.lua:341: attempt to index a nil value (local 'f')
```

**Comment by cedounet** [comment]
:/  Let me have a look.

**Comment by cedounet** [comment]
Yeah that is something wrong with make4ht/tex4ebook

the-sourdough-framework/book ➤ 9611b25|list_of_flowchart⚡
3814 ± : head book_serif/book.loc                                                                                          [-28208758m] ✹ ✭
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\contentsline {flowchart}{\numberline {1}{\ignorespaces Process for starter from scratch}}{19}{flowchart.caption.14}%
\contentsline {flowchart}{\numberline {2}{\ignorespaces Sourdough readiness flow chart}}{22}{flowchart.caption.16}%
\contentsline {flowchart}{\numberline {3}{\ignorespaces Sourdough starter maintenance flowchart}}{24}{flowchart.caption.17}%
\addvspace {10\p@ }

it is all empty with maek4ht
book ➤ 9611b25|list_of_flowchart⚡
3815 ± : head book.loc                                                                                                     [-28208758m] ✹ ✭
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }
\addvspace {10\p@ }

**Comment by cedounet** [comment]
> Good point. The CI doesn't run trying to make the ebook version. This seems to be broken too. Got this error when running `make bake`
> 
> ```
> [WARNING] domfilter: ...ive/2023/texmf-dist/tex/luatex/luaxml/luaxml-mod-xml.lua:175: Unbalanced Tag (/body) [char=15766]
> [...]
> [WARNING] exec_epub: Missing opf file
> ...023/texmf-dist/scripts/tex4ebook/tex4ebook-exec_epub.lua:341: attempt to index a nil value (local 'f')
> ```

FWIW local builds work on my debian machine.

Our website issue is related to the .loc file being empty.

**Comment by cedounet** [comment]
Some more debug info
the file book.4ct has loc, lof and lot infos..

My google-fu is failing me right now.

>  grep loc book.4ct | head
\doTocEntry\tocloc{1}{\csname a:TocLink\endcsname{7}{x7-14002r1}{}{\ignorespaces Process for starter from scratch}}{flowchart}\relax 
\doTocEntry\tocloc{2}{\csname a:TocLink\endcsname{7}{x7-15001r2}{}{\ignorespaces Sourdough readiness flow chart}}{flowchart}\relax 
\doTocEntry\tocloc{3}{\csname a:TocLink\endcsname{7}{x7-16001r3}{}{\ignorespaces Sourdough starter maintenance flowchart}}{flowchart}\relax 
\doTocEntry\tocloc{4}{\csname a:TocLink\endcsname{10}{x10-27001r4}{}{\ignorespaces Flat bread process}}{flowchart}\relax 
\doTocEntry\tocloc{5}{\csname a:TocLink\endcsname{11}{x11-29001r5}{}{\ignorespaces The typical process of making a wheat-based sourdough bread.\relax }}{flowchart}\relax 
\doTocEntry\tocloc{6}{\csname a:TocLink\endcsname{11}{x11-30001r6}{}{\ignorespaces Process to prepare your starter before baking}}{flowchart}\relax 
\doTocEntry\tocloc{7}{\csname a:TocLink\endcsname{11}{x11-36001r7}{}{\ignorespaces The gluten development process for a wheat-based dough.\relax }}{flowchart}\relax 
\doTocEntry\tocloc{8}{\csname a:TocLink\endcsname{11}{x11-37002r8}{}{\ignorespaces Process to check the bulk fermentation}}{flowchart}\relax 
\doTocEntry\tocloc{9}{\csname a:TocLink\endcsname{11}{x11-39001r9}{}{\ignorespaces Dividing decision tree}}{flowchart}\relax 
\doTocEntry\tocloc{10}{\csname a:TocLink\endcsname{11}{x11-40001r10}{}{\ignorespaces Shaping process}}{flowchart}\relax

**Comment by hendricius** [comment]
Maybe I can reach out to Michal?

**Comment by cedounet** [comment]
Was planning to Google some more and if not do a MWE before reaching out.

**Comment by cedounet** [comment]
OK here my MWE if you could confirm it "works" (i.e. fails) for you too:
`\documentclass{scrbook}
\DeclareNewTOC[
  type=flowchart,%
  types=flowcharts,%  ,
  float,
  name=Flowchart,%
  listname = {List of Flowcharts}
]{loc}
\setuptoc{loc}{chapteratlist}

\begin{document}
\begin{flowchart}[!htb]
    \caption{A flowchart}
\end{flowchart}

\begin{figure}[!htb]
    \caption{A figure}
\end{figure}
\listofflowcharts
\listoffigures

\end{document}`

If i do `lualatex mwe.tex` then `cat mwe.loc` I get:
 `\contentsline {flowchart}{\numberline {1}{\ignorespaces A flowchart}}{1}{}%
\providecommand \tocbasic@end@toc@file {}\tocbasic@end@toc@file 
`
and the pdf is as expected

If I do ` make4ht --lua -a debug mwe.tex` then `cat mwe.loc` I get an empty file:
`\providecommand \tocbasic@end@toc@file {}\tocbasic@end@toc@file 
`

**Comment by hendricius** [comment]
@cedounet great test. Can confirm - I get the same error.

**Comment by cedounet** [comment]
raised https://github.com/michal-h21/tex4ebook/issues/114

**Comment by cedounet** [comment]
Updated with michal's comments. @hendricius please have a look if that works (reverted your workaround).  My ruby install is flaky right now.

**Comment by hendricius** [comment]
@cedounet thank you!!!

For some reason after trying - I still don't see the list of tables in the navigation bar on the right. However - the file is generated. I will add a workaround for this for now. I will merge the PR and create a followup PR. Side note - could you maybe directly create your branches on this repo? Then I can push to your PRs and/or you can push to mine. Thank you!

**Comment by cedounet** [comment]
> Side note - could you maybe directly create your branches on this repo? 
> Then I can push to your PRs and/or you can push to mine. Thank you!

I will try yes

**Pull request merged**